assets/app/scripts/controllers/build.js

```diff
@@ -99,8 +99,34 @@ angular.module('openshiftConsole')
           if (!$filter('isIncompleteBuild')(build)){
             delete $scope.buildConfigBuildsInProgress[buildConfigName][buildName];
           }
-        }        
+        }
       }));
+
+      (function initLogs() {
```

Comment:
It runs the code, but also lets me pass `initLogs` as a named function to the view so I can let the user retry if something goes wonky:

```javascript
(function initLogs() {
        angular.extend($scope, {
          logs: [],
          logsLoading: true,
          canShowDownload: false,
          initLogs: initLogs  <-- it ran once, but can run again via user 
        });
```
and in the view:

```html
<a ng-click="initLogs()">reconnect</a>
```